### PR TITLE
Improve integration tests

### PR DESCRIPTION
**Pull request by vitaliystoliarovcc**
- added tests for OCOCO, Ping/Pong, Acc/Dist and Iceberg ago orders
- fixed iteration of incoming/outgoing events in SessionSpy
- reduced interval of streaming to channel
- added Trades data provider
- fixed Book data provider
- extended BitfinexSessionMock to allow emulate outgoing events from the tests

### CODE REVIEW

File: test/integration/accumulate_distribute.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    spyConn
+      .received('subscribe', (e) => expect(e.channel).to.eq('book'))
+      .sent('subscribed', (e) => {
+        expect(e.channel).to.eq('book')
+        expect(e.chanId).to.be.a('number')
+      })
+      .received(NEW_ORDER, ({ fields: [placeholder, details] }) => {
+        expect(details.symbol).to.eq('tAAABBB')
+        expect(details.type).to.eq('EXCHANGE LIMIT')
+        expect(+details.amount).to.be.within(-0.2, -Number.EPSILON)
```
Comment (avsek477) [inline test/integration/accumulate_distribute.js:72]: makes more accurate if the range is between -1.2 to -0.8 according to the given input

**Review by avsek477 — APPROVED** [review]
rest lgtm

**Review by tarcisiozf — APPROVED** [review]
looks great

File: test/util/base_api_mock.js [hunk 2]
```diff
@@ -39,11 +39,20 @@ class BaseApiMock extends ApiMock {
       low: 9_000,
       volume: 3
     })
+    const baseTrade = new PublicTrade({ price: 5000, amount: 9001 })
+    const baseBook = new OrderBook([
+      [67536333723, 500, -2],
+      [67538314922, 600, -2],
+      [67535007022, 500, 2],
```
Comment (robertkowalski) [inline test/util/base_api_mock.js:46]: both heads at `500`, wouldn't they have matched?

Comment (robertkowalski) [reply test/util/base_api_mock.js:46]: should be `499` for the buy instead?

File: test/util/mimic/data-providers/trades.js [hunk 3]
```diff
@@ -0,0 +1,18 @@
+'use strict'
+
+// const timeFrames = require('bfx-hf-util/lib/candles/time_frames')
+// const parseChannelKey = require('../../../../lib/util/parse_channel_key')
```
Comment (robertkowalski) [inline test/util/mimic/data-providers/trades.js:4]: commented? should we remove?

File: test/util/mimic/responses/order.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
+module.exports = function ({ id, gid, cid, amount, status }) {
```
Comment (robertkowalski) [inline test/util/mimic/responses/order.js:1]: use strict

File: test/util/base_api_mock.js (hunk 2 above)
Comment (vitaliystoliarovcc) [reply test/util/base_api_mock.js:46]: Yeah, it can, but it seems that in this situation it does not affect

**Review by robertkowalski — APPROVED** [review]
lgtm, will merge for 3.14

**Pull request merged**